test/scheduler/actuals_test.go

```diff
@@ -0,0 +1,405 @@
+
+		// Check that the returned list is empty.
+		if bindingCount := len(bindingList.Items); bindingCount != 0 {
+			return fmt.Errorf("%d bindings have been created unexpectedly", bindingCount)
+		}
+
+		return nil
+	}
+)
+
+func crpSchedulerFinalizerAddedActual(crpName string) func() error {
+	return func() error {
+		// Retrieve the CRP.
+		crp := &fleetv1beta1.ClusterResourcePlacement{}
+		if err := hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp); err != nil {
+			return err
+		}
+
+		// Check that the scheduler finalizer has been added.
+		if !controllerutil.ContainsFinalizer(crp, fleetv1beta1.SchedulerCRPCleanupFinalizer) {
+			return fmt.Errorf("scheduler cleanup finalizer has not been added")
+		}
+
+		return nil
+	}
+}
+
+func crpSchedulerFinalizerRemovedActual(crpName string) func() error {
+	return func() error {
+		// Retrieve the CRP.
+		crp := &fleetv1beta1.ClusterResourcePlacement{}
+		if err := hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp); err != nil {
+			return err
+		}
+
+		// Check that the scheduler finalizer has been added.
+		if controllerutil.ContainsFinalizer(crp, fleetv1beta1.SchedulerCRPCleanupFinalizer) {
+			return fmt.Errorf("scheduler cleanup finalizer is still present")
+		}
+
+		return nil
+	}
+}
+
+func scheduledBindingsCreatedForClustersActual(clusters []string, scoreByCluster map[string]*fleetv1beta1.ClusterScore, crpName, policySnapshotName string) func() error {
+	return func() error {
+		// List all bindings.
+		bindingList := &fleetv1beta1.ClusterResourceBindingList{}
+		if err := hubClient.List(ctx, bindingList); err != nil {
+			return err
+		}
+
+		// Find all the scheduled bindings.
+		scheduled := []fleetv1beta1.ClusterResourceBinding{}
+		for _, binding := range bindingList.Items {
+			if binding.Spec.State == fleetv1beta1.BindingStateScheduled {
+				scheduled = append(scheduled, binding)
+			}
+		}
+
+		// Verify that scheduled bindings are created as expected.
+		wantScheduled := []fleetv1beta1.ClusterResourceBinding{}
+		for _, name := range clusters {
+			score := scoreByCluster[name]
+			binding := fleetv1beta1.ClusterResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: bindingNamePlaceholder,
+					Labels: map[string]string{
+						fleetv1beta1.CRPTrackingLabel: crpName,
+					},
+				},
+				Spec: fleetv1beta1.ResourceBindingSpec{
+					State:                        fleetv1beta1.BindingStateScheduled,
+					SchedulingPolicySnapshotName: policySnapshotName,
+					TargetCluster:                name,
+					ClusterDecision: fleetv1beta1.ClusterDecision{
+						ClusterName:  name,
+						Selected:     true,
+						ClusterScore: score,
+					},
+				},
+			}
+			wantScheduled = append(wantScheduled, binding)
+		}
+
+		if diff := cmp.Diff(
+			scheduled,
+			wantScheduled,
+			ignoreResourceBindingTypeMetaField,
+			ignoreObjectMetaNameField,
+			ignoreObjectMetaAutoGeneratedFields,
+			ignoreClusterDecisionReasonField,
+			cmpopts.SortSlices(lessFuncBinding),
```

Comment:
```suggestion
                options := []cmp.Option{
				ignoreResourceBindingTypeMetaField,
			        ignoreObjectMetaNameField,
			        ignoreObjectMetaAutoGeneratedFields,
			        ignoreClusterDecisionReasonField,
			        cmpopts.SortSlices(lessFuncBinding),
			}
		if diff := cmp.Diff(scheduled, wantScheduled, options...),
```
refactor it as a common variable? it is used in multiple places

Reply:
Done.